docs/proposals/registry/devfile-packaging.md

```diff
@@ -0,0 +1,53 @@
+
+
+Each folder under devfiles corresponds to a stack, with its own devfile and associated artifacts. There may be an index.json present for devfile consumers
+
+From this point, the Github repository may act as self-hosting devfile registry (as in the case of [odo-devfiles/registry](https://github.com/odo-devfiles/registry), or it may be built up into an Apache server container (as in the case of v1 registries like [eclipse/che-devfile-registry](https://github.com/eclipse/che-devfile-registry)).
+
+## Are any changes needed to accommodate OCI?
+Storing the devfiles (and any associated artifacts) on a dedicated Github repository (the “devfile registry repository”), is still a good idea.
+
+   - Provides a clear, straightforward and non-cluttered place for stack owners to contribute devfiles to
+   - Separates the devfile stacks from the OCI registry code and logic
+
+But, with the move to OCI registries, we need a way to easily package the devfile stacks from GitHub and load them into the OCI registry
+
+   - Needs to retrieve the devfiles without an internet connection
+     - Means we can’t have the bootstrap container git clone the devfiles before pushing them up
+   - Needs to be easily built and distributable by registry admins.  
+     - Once we support “deploying your own registry”, with custom devfiles this will become especially important
+   - Shouldn’t require building multiple components of the registry to distribute
+
+## Proposed Changes
+
+**To address the requirements listed above, have the “devfile registry repository” built into a custom bootstrap image, based on a base-image provided by us.**
+
+<img width="633" alt="Screen Shot 2020-10-29 at 12 25 53 PM" src="https://user-images.githubusercontent.com/6880023/97602641-eaa5c300-19e1-11eb-8c3b-1c3ee0cb6f11.png">
+
+
+The base-image will be the "bootstrap container" image that's currently in the [devfile/registry-support](https://github.com/devfile/registry-support) repo and will be hosted on the `devfile` org on quay.io. It contains the logic for loading the devfiles into the OCI registry as well as hosting the index.json.
+
+The Dockerfile will just contain lines to copy the devfiles and index.json into the container, nothing else will be required (unless the registry administrator chooses to add more).
+
+As part of the registry build in the registry repository (such as [devfile/registry](https://github.com/devfile/registry)), the index.json will be generated and any tests/validation specific to the repository will be run. After this, the Dockerfile will be built and an image will be produced and pushed up to a Docker registry.
+
+When deploying the OCI registry, the registry's image will be specified as the bootstrap container. Then, when the bootstrap container starts, it pushes the devfiles up into the OCI registry using `oras` and serves the pre-generated index.json.
```

Comment:
> After this, the Dockerfile will be built and an image will be produced and pushed up to a Docker registry.

Another question is about the line above, the `Dockerfile` refers to which Dockerfile? Is it the Dockerfile is the one admin builds and it is based on the "bootstrap image"? I would be better we have a terminology section in this design proposal to define the terminology (eg. "bootstrap container", "devfile registry repository", "registry Dockerfile", "registry support repository", etc) and we can use that in the following content.

Reply:
Yup, it's the registry's Dockerfile. Sure, I can add a terminology section.

Reply:
Thanks John!

Reply:
@GeekArthur Updated to add a terminology section 👍